packages/ag-charts-community/src/options/chart/chartBuilderOptions.ts

```diff
@@ -24,3 +29,62 @@ export interface AgChartInstance {
     /** Destroy the chart instance and any allocated resources to support its rendering. */
     destroy(): void;
 }
+
+export interface AgChartInterface {
+    /**
+     * Create a new `AgChartInstance` based upon the given configuration options.
+     */
+    create(options: AgChartOptions & AgChartSpecialOverrides): AgChartInstance;
+
+    /**
+     * Update an existing `AgChartInstance`. Options provided should be complete and not
+     * partial.
+     * <br/>
+     * <br/>
+     * **NOTE**: As each call could trigger a chart redraw, multiple calls to update options in
+     * quick succession could result in undesirable flickering, so callers should batch up and/or
+     * debounce changes to avoid unintended partial update renderings.
+     */
+    update(chart: AgChartInstance, options: AgChartOptions & AgChartSpecialOverrides): void;
+
+    /**
+     * Update an existing `AgChartInstance` by applying a partial set of option changes.
+     * <br/>
+     * <br/>
+     * **NOTE**: As each call could trigger a chart redraw, each individual delta options update
+     * should leave the chart in a valid options state. Also, multiple calls to update options in
+     * quick succession could result in undesirable flickering, so callers should batch up and/or
+     * debounce changes to avoid unintended partial update renderings.
+     */
+    updateDelta(chart: AgChartInstance, deltaOptions: AgChartOptions): void;
+
+    /**
+     * Starts a browser-based image download for the given `AgChartInstance`.
+     */
+    download(chart: AgChartInstance, options?: DownloadOptions): void;
+
+    /**
+     * Returns a base64-encoded image data URL for the given `AgChartInstance`.
+     */
+    getImageDataURL(chart: AgChartInstance, options?: ImageDataUrlOptions): Promise<string>;
+}
+
+export interface AgChartSpecialOverrides {
```

Comment:
This shouldn't be public - can we keep it internal, it's mostly for testing and SSR cases?

Reply:
Oh, good to know. I'll fix it 👍🏼